brave/brave-ios
# No Bug: Create BraveRewardsUI framework target and move brave-rewards-ios there

This moves the contents of brave-rewards-ios into its own framework target (and unit test bundle) within brave-ios/Client.

**One thing to note**: we will need to figure out how to either separate the transifex strings for rewards (as these are bundled within the BraveRewardsUI framework) or move/merge them into BraveShared/BraveStrings.swift and update BraveRewardsUI to grab them from there.

## Submitter Checklist:


## Test Plan:

- Enable rewards by removing NO_REWARDS flag
- Make sure to remove the Carthage build for brave-rewards-ios (ensuring it is not linking/copying the previous version)
- Sanity test that rewards works within brave-ios

## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
